json-schema-provider/src/main/java/io/confluent/kafka/schemaregistry/json/diff/SchemaDiff.java

```diff
@@ -208,6 +220,34 @@ static void compare(final Context ctx, Schema original, Schema update) {
     }
   }
 
+  private static List<Difference> compareMetadata(Schema update,
+                                                  Map<String, String> originalProperties,
+                                                  Map<String, String> updatedProperties) {
+    List<Difference> differences = new ArrayList<>();
+    if (update instanceof ObjectSchema) {
+      ObjectSchema updatedObjectSchema = (ObjectSchema) update;
+      Set<String> originalReservedPropertyKeys =
+              Sets.newHashSet(originalProperties.getOrDefault(JsonSchema.RESERVED, "").split(","));
+      Set<String> updatedPropertyKeys = updatedObjectSchema.getPropertySchemas().keySet();
+      Set<String> updatedReservedPropertyKeys =
+              Sets.newHashSet(updatedProperties.getOrDefault(JsonSchema.RESERVED, "").split(","));
+      // backward compatibility check to ensure that original reserved properties are not removed in
+      // the updated version
+      if (!Sets.difference(originalReservedPropertyKeys, updatedReservedPropertyKeys).isEmpty()) {
+        differences.add(new Difference(Type.RESERVED_PROPERTY_REMOVED, ""));
```

Comment:
If we create a RESERVED_PROPERTY_REMOVED for each one, then you can use the path here.